portable/MemMang/heap_4.c

```diff
@@ -61,7 +61,7 @@ task.h is included from an application file. */
 	heap - probably so it can be placed in a special segment or address. */
 	extern uint8_t ucHeap[ configTOTAL_HEAP_SIZE ];
 #else
-	static uint8_t ucHeap[ configTOTAL_HEAP_SIZE ];
+	PRIVILEGED_DATA static uint8_t ucHeap[ configTOTAL_HEAP_SIZE ];

```

Comment:
What impact will this have on existing applications where the heap is unprivileged?

Reply:
The only possible impact is on the unprivileged software as the privileged software already has RW access to privileged data region. 

**Use Case** - An application allocates heap memory and uses it in an unprivileged task. Note that it is only possible to do so from the privileged software at the time of creation as the unprivileged task cannot call `pvPortMalloc` because there is no mpu wrapper mapping: https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/master/include/mpu_wrappers.h.

The application must already be using an MPU region to grant the task access to the memory allocated using heap. [This is achievable in v8M but seems more of a theoretical use case for v7M - it is unreasonable to expect that a block of memory allocated using `pvPortMalloc` will satisfy all the MPU constraints.]

**Impact**
* v7M - This will continue to work as overlapping regions are allowed and the highest region number takes precedence. However, because of MPU restrictions, I do not think it is possible to expect that a block of memory allocated from heap will satisfy all the MPU restrictions and therefore, it is more of a re
* v8M - This will not work as overlapping regions are not allowed.